### PR TITLE
[ENDPOINT]: Created endpoint to make transactions.

### PR DESCRIPTION
**Pull request by marce-alf**
## Jira Ticket

- [WA-19]

## Description
Created endpoint and middleware to validate and create transactions.

AS: a user
I Want: the ability to do transactions
To: make movements.

## Evidence:
Transaction when all conditions are met:
create-transaction

Transaction when giving amount as 0
transaction-validation

Transaction when giving an incorrect date:
transaction-validation-2

Transaction when giving a non-valid user:
transaction-validation-3

### CODE REVIEW

File: controllers/transaction.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+      endpointResponse({
+        res,
+        message: 'Transaction created successfully.',
+        body,
+      });
+    } catch (error) {
+      const httpError = createHttpError(
+        error.statusCode,
+        `[Error retrieving index] - [index - GET]: ${error.message}`
```
Comment (ledelaespriella) [inline controllers/transaction.js]: correct error description.

> [Error retrieving Transaction] - [Transaction - POST]:

File: routes/transaction.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const express = require('express');
+const { postCreateTransaction } = require('../controllers/transaction');
+const postTransactionSchema = require('../schemas/transaction/postTransactionSchema');
+const validator = require('../middlewares/validator');
+const router = express.Router();
+
+router.post('/', validator(postTransactionSchema), postCreateTransaction);
```
Comment (ledelaespriella) [inline routes/transaction.js:7]: You must create a folder called transactions and inside place the js of the route.

Captura de Pantalla 2022-11-03 a la(s) 8 04 19 p m

File: routes/index.js [hunk 3]
```diff
@@ -5,5 +5,6 @@ const router = express.Router()
 
 // example of a route with index controller get function
 router.get('/', get)
+router.use('/transactions', transactionRoutes);
```
Comment (ledelaespriella) [inline routes/index.js:8]: Delete line 8

**Review by ledelaespriella — CHANGES_REQUESTED** [review]
## Pending

Attach evidence of how the return would be when a parameter is passed that does not meet the conditions or is missing

File: routes/transaction.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+const { postCreateTransaction } = require('../controllers/transaction');
+const postTransactionSchema = require('../schemas/transaction/postTransactionSchema');
+const {
+    validateRequestSchema,
+  } = require('../middlewares/validation/validate-schema.middleware');
+const router = express.Router();
+
+router.post('/', validateRequestSchema(postTransactionSchema), postCreateTransaction);
+
+module.exports = router;
```
Comment (ledelaespriella) [inline routes/transaction.js]: rename file to: transactions.routes.js

File: controllers/transaction.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+const createHttpError = require('http-errors');
+const { Transaction } = require('../database/models');
+const { endpointResponse } = require('../helpers/success');
+const { catchAsync } = require('../helpers/catchAsync');
+
+// example of a controller. First call the service, then build the controller method
+module.exports = {
```
Comment (ledelaespriella) [inline controllers/transaction.js]: rename file to: transactions.controller.js

**Review by ledelaespriella — CHANGES_REQUESTED** [review]
### **Pending**

change file names

**Review by ledelaespriella — APPROVED** [review]
All approved

**Review by germanWalton — APPROVED** [review]

**Pull request merged**